app/dashboard/models.py

```diff
@@ -1151,15 +1156,59 @@ def can_remarket(self):
             result = False
 
         if self.last_remarketed:
-            one_hour_after_remarketing = self.last_remarketed + timezone.timedelta(hours=1)
-            if timezone.now() < one_hour_after_remarketing:
+            minimum_wait_after_remarketing = self.last_remarketed + timezone.timedelta(minutes=settings.MINUTES_BETWEEN_RE_MARKETING)
+            if timezone.now() < minimum_wait_after_remarketing:
                 result = False
 
         if self.interested.count() > 0:
             result = False
 
         return result
 
+    @property
+    def is_reserved(self):
+        if self.bounty_reserved_for_user and self.reserved_for_user_from:
+            if timezone.now() < self.reserved_for_user_from:
+                return False
+
+            if self.reserved_for_user_expiration and timezone.now() > self.reserved_for_user_expiration:
+                return False
+
+            return True
+
+    @property
+    def total_reserved_length_label(self):
+        if self.bounty_reserved_for_user and self.reserved_for_user_from:
+            if self.reserved_for_user_expiration is None:
+                return 'indefinitely'
+
+            if self.reserved_for_user_from == self.reserved_for_user_expiration:
+                return ''
+
+            delta = self.reserved_for_user_expiration - self.reserved_for_user_from
+            days = delta.days
+
+            if days > 0:
```

Comment:
Is there a simpler way we can humanize the timedelta?  https://anandnalya.com/2009/05/humanizing-the-time-difference-in-django/

Reply:
Hey Dan. I'm not sure what you mean? I've had a look at the link you sent and we would still need to do calcs for hours. Additionally, the humanised timedelta outputs seem different compared to the current requirement of allowing a funder to reserve for '3 days', '1 week' or 'Keep reserved'. I tried to make the code as scalable as possible and struggled to find easier ways of doing this. It's a shame the timedelta object doesn't have an hours getter attribute or even a weeks one which is strange as you can construct a timedelta with weeks, days, hours, minutes etc.

Reply:
@danlipert

Reply:
@danlipert any thoughts on my comment?